feat: introduce log content on log records

AWS-Samuel opened the pull request:
### What was the problem/requirement? (What/Why)

Consumers of the `openjd_sessions` module may want a means of filtering logs based on their contents. 

### What was the solution? (How)

In order to solve this problem, this commit introduces a structure we can add to the `extra` kwarg when logging messages. The value provided to the `extra` kwarg must be of type `dict[str, Any]`, with the limitation that the keys cannot be an [existing attribute name in LogRecord](https://docs.python.org/3/library/logging.html#logrecord-attributes).

This commit then introduces a  `LogExtraInfo`  class which is a Dictionary containing one key: `openjd_log_content`.

`openjd_log_content` is a `Flag` which allows combining multiple `LogContent` values through bitwise operators in order to indicate which contents are in the logged message. 

### What is the impact of this change?

A `LogContent` flag becomes available to attached to logged messages, that consumers of this package can use to filter log messages based on the type of log message.

### How was this change tested?

> Have you run the unit tests?

Yes, also added unit tests to confirm all expected fields in log records still exist

### Was this change documented?

It is through the changelog. 
  
### Is this a breaking change?

No

### Does this change impact security?

No

----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

AWS-Samuel committed:
feat: introduce log content on log records

Signed-off-by: Samuel Anderson <119458760+AWS-Samuel@users.noreply.github.com>

ddneilson commented on src/openjd/sessions/_logging.py:
```diff
@@ -1,8 +1,45 @@
+    log_content: LogContent = LogContent(0)  # Empty flag
+
+
+class OJDExtraInfo(TypedDict):
```
nit: I think that the `OJD` prefix is a given, since it's a private class within the `openjd-sessions` module.

```suggestion
class LogExtraInfo(TypedDict):
```

AWS-Samuel replied:
Amended

ddneilson reviewed (changes requested):
This is great, Sam. Thanks for taking this on.

Can you also add some public documentation for this so that users of the library can discover that this extra log metadata exists? Perhaps as a docstring on the `LOG` global in this file?

For example, we communicate that there's a session_id metadata in the log messages [here](https://github.com/OpenJobDescription/openjd-sessions-for-python/blob/009737a6a02b9c048085949c85e036fa0aa210e3/src/openjd/sessions/_session.py#L178-L182).

ddneilson commented:
Chatted offline. We'll also need to make sure that the uses of this don't clobber the `extra` that is added by the [`Session`'s `LoggerAdaptor`](https://github.com/OpenJobDescription/openjd-sessions-for-python/blob/009737a6a02b9c048085949c85e036fa0aa210e3/src/openjd/sessions/_session.py#L397). All of the logger instances that are used in this library are actually instances of that object, so we'll need to make sure that it still works as expected.

AWS-Samuel commented:
> This is great, Sam. Thanks for taking this on.
> 
> Can you also add some public documentation for this so that users of the library can discover that this extra log metadata exists? Perhaps as a docstring on the `LOG` global in this file?
> 
> For example, we communicate that there's a session_id metadata in the log messages [here](https://github.com/OpenJobDescription/openjd-sessions-for-python/blob/009737a6a02b9c048085949c85e036fa0aa210e3/src/openjd/sessions/_session.py#L178-L182).

Added, confirmed that the documentation shows up under hover text for `LOG` in my editor.

>Chatted offline. We'll also need to make sure that the uses of this don't clobber the extra that is added by the [Session's LoggerAdaptor](https://github.com/OpenJobDescription/openjd-sessions-for-python/blob/009737a6a02b9c048085949c85e036fa0aa210e3/src/openjd/sessions/_session.py#L397). All of the logger instances that are used in this library are actually instances of that object, so we'll need to make sure that it still works as expected.

Confirmed that this was causing issue, so I added an override and added a test for this.

ddneilson commented on src/openjd/sessions/_logging.py:
```diff
@@ -1,22 +1,89 @@
 LOG = logging.getLogger(".".join(__name__.split(".")[:-1]))
+"""
+The logger of the openjd sessions module. The logger has the name openjd.sessions and is used 
+throughout the openjd sessions module to provide information on actions OpenJD is taking, as well as
```
```suggestion
throughout the openjd sessions module to provide information on actions the module is taking, as well as
```

AWS-Samuel replied:
Thanks, added in latest

ddneilson reviewed (changes requested):
One small nitpick. Thanks for fixing this up, Sam.
